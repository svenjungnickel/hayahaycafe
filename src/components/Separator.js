import React from 'react';
import PropTypes from 'prop-types';
import { separator } from '../styles/components/Separator.module.scss';

const Separator = ({ className }) => (
    <div className={'' !== className ? `${separator} ${className}` : `${separator}`}>
        <hr />
    </div>
);

Separator.defaultProps = {
    className: '',
};

Separator.propTypes = {
    className: PropTypes.string,
};

export default Separator;
