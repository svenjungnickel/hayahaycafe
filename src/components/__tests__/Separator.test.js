import React from 'react';
import renderer from 'react-test-renderer';
import Separator from '../Separator';

describe('Separator', () => {
    it('renders without className', () => {
        const component = <Separator />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders with className', () => {
        const className = 'className';

        const component = <Separator className={className} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
