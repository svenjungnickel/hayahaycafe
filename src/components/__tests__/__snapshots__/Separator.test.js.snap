// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Separator renders with className 1`] = `
<div
  className="separator className"
>
  <hr />
</div>
`;

exports[`Separator renders without className 1`] = `
<div
  className="separator"
>
  <hr />
</div>
`;
